lib/recurly/pricing/index.js

```diff
@@ -243,14 +252,18 @@ Pricing.prototype.coupon = function (couponCode, done) {
 
   return new PricingPromise(function (resolve, reject) {
     if (!self.items.plan) return reject(errors('missing-plan'));
-    if (coupon) {
-      if (coupon.code === couponCode) return resolve(coupon);
-      else self.remove({ coupon: coupon.code });
+    if (coupon) self.remove({ coupon: coupon.code });
```

Comment:
This ensures we always process coupon set calls, since plans may have changed between calls. I don't think we should be tracking plan change histories in the coupon method.